### ISSUE TITLE
Treat minimized windows as closed

### ISSUE THREAD

**Issue by cassidyjames** [elementary/gala#740]
I feel like minimized windows are in this weird in between state where we keep them on a workspace and show them in the multitasking overview, but then when switching to a workspace they disappear. I wonder if we could deprioritize them further by basically treating them as closed, by:

1. Not showing in workspace overviews
2. Not showing in Alt+Tab window switching
3. Effectively making them on all workspaces, so restoring a minimized window would work no matter which workspace you're on.

I think this would actually make minimizing more useful as a pseudo-close, while also making it less in conflict with how we've designed window management. It moves minimizing closer to closing a window while still retaining the underlying functionality.

**Comment by Blast-City** [comment]
Wouldn't it be better if minimize was removed from dock, context menus, keyboard shortcuts, etc and leave the gsettings to people who will complain?

**Comment by donadigo** [comment]
I wrote this previously on Slack, but I think it could be valuable here:
So after thinking for a while, the issue above seems to be still treating minimize as a separate action but when it comes to how system will represent this state, it isn't actually that far off from being completely closed. I feel like what we are really coming for is "close = app is still in memory but it isn't visible anywhere in the OS and when you launch it again, you're just making it visible again, just like unminimizing". This essentially makes minimizining unnecessary.

If this behaviour got implemented, we would have two very similar actions that don't really provide any benefits over them anymore.

**Comment by cassidyjames** [comment]
@donadigo I've previously wondered if we should remove close and actually have it be minimize, but this seemed like a smaller step toward that sort of thinking that would be achievable today, and without massive implications for how all apps and memory management work. We've already mostly deprecated minimizing, but as @Blast-City mentioned, some apps do it themselves. We could treat it more like a closed state today, then decide in the future if we really want to completely deprecate it or to deprecate "close" in favor of minimize, or what.

But regardless of what we decide to do in the future, I think this is a step in the right direction.

**Comment by Blast-City** [comment]
Just a thing that I thought the other day: 

Usually users don't make Steam games icons on the dock. If a game minimizes by it self when unfocused and the minimized game doesn't show on window switcher, workspace overview and the dock, how would an user restore the game again? Is there an option to restore a game window on the Steam main window?

The same goes for apps that aren't pinned on dock. It is a little bit tedious to go to the application menu every time a screenrecorder minimizes because it is recording and that option is not even possible on the Steam scenario.

Am I missing something?

**Comment by cassidyjames** [comment]
@Blast-City ah I guess I was thinking Plank would still show those icons, maybe. Because the app is running, just not visible. But yeah maybe that's too mixed of experiences?

**Comment by Blast-City** [comment]
> @Blast-City ah I guess I was thinking Plank would still show those icons, maybe. Because the app is running, just not visible. But yeah maybe that's too mixed of experiences?

Yeah, that's the problem. It would still be three states: Closed, Maximized, Minimized rebranded. I guess that an user would think why is there a application on the dock that is not docked or have a running indicator.

I think the idea is to have just two states.

**Comment by cassidyjames** [comment]
@Blast-City at the same time, we do show apps on the dock that have no open windows but are showing a dock badge or progress bar. Maybe that's enough of a concession for those long-running apps; minimizing ideally is not a thing apps do to themselves, anyway.

**Comment by cassidyjames** [comment]
@danrabbit this was this issue I mentioned earlier today in our call.

**Comment by Julioevm** [comment]
Hey guys,

Do you have in consideration the huge amount of people that actively use the minimize function? One thing is not having a minimize icon on the window, and other thing is changing how minimizing works. I feel like this is taking into account the use case of a very few people, because a change like this in my case, and I'm guessing, many others would impact considerably my workflow, in a negative way.

Maybe if you use exclusively full screen applications and always focus on one window, this could make some sense, but this is not the case many times, specially if you use the os on a bigger monitor than the usual laptop size.

I do want to minimize windows so that they're not taking screen space momentarily, juggling around workspaces for something I want to hide a few minutes is unintuitive and not very practical. And I expect to see a minimized window in the multitasking overview, because its a window I'm working with, I just want to have it out of sight for a while, but I need to have it localized for when I need it back. If I want a window out of my current flow and don't wanna close it, I just send it to a different workspace, I find that more useful.

What exactly is wrong with a decades old feature that many still use, and otherwise doesn't harm in any way how a desktop behaves?
I invite you to reconsider how this change would affect many users, and what benefits does it really bring. 

Thanks,

**Comment by lenemter** [comment]
Closing because it doesn't make sense to introduce more code to discourage a feature we don't even support
